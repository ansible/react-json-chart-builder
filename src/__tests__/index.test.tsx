import React from 'react'
import { render, waitFor } from '@testing-library/react';
import fetchMock from 'fetch-mock-jest';
import {
    stackedChartResponse,
    pieChartResponse
} from './assets/fetchResponses';
import ChartRenderer from '../components/';
import {
    ChartKind,
    ChartTopLevelType,
    functions,
    ChartSchemaElement,
    ChartType
} from '../index';

const charts: ChartSchemaElement[] = [
    {
        id: 1000,
        kind: ChartKind.wrapper,
        type: ChartTopLevelType.chart,
        parent: null,
        props: {
            height: 300
        },
        xAxis: {
            label: 'Date',
        },
        yAxis: {
            label: 'Jobs across all clusters'
        },
        api: {
            params: {},
            url: 'http://chart1.url'
        }
    },
    {
        id: 1001,
        kind: ChartKind.simple,
        type: ChartType.line,
        parent: 1000,
        props: {
            x: 'created_date',
            y: 'success_count'
        }
    }
];


describe('Chart/Renderers/CreateWrapper', () => {
    afterEach(() => fetchMock.restore());

    test('should render one chart specified in the id', async () => {
        fetchMock.post({ url: 'http://chart1.url' }, stackedChartResponse);

        const { container } = render(<ChartRenderer
            schema={charts}
            functions={functions}
        />);
        await waitFor(() => container.querySelector('svg'));

        expect(container).toMatchSnapshot();

    });
});