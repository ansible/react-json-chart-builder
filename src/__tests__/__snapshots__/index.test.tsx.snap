// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart/Renderers/CreateWrapper should render one chart specified in the id 1`] = `
<div>
  <div>
    <div
      style="height: 300px;"
    >
      <div
        class="pf-c-chart"
        style="height: 100%; width: 100%; user-select: none; pointer-events: none; position: relative;"
      >
        <svg
          height="300"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 500 300"
          width="500"
        >
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #d2d2d2; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="70"
              x2="490"
              y1="230"
              y2="230"
            />
            <text
              direction="inherit"
              dx="0"
              id="chart-axis-0-axisLabel-0"
              x="280"
              y="281.97"
            >
              <tspan
                dx="0"
                dy="0"
                style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 40px; stroke: transparent; text-anchor: middle; fill: #252525;"
                text-anchor="middle"
                x="280"
              >
                Date
              </tspan>
            </text>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="70"
                x2="70"
                y1="230"
                y2="235"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-0"
                x="70"
                y="256.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="middle"
                  x="70"
                >
                  2020-10-31
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="238"
                x2="238"
                y1="230"
                y2="235"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-2"
                x="238"
                y="256.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="middle"
                  x="238"
                >
                  2020-11-02
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="406"
                x2="406"
                y1="230"
                y2="235"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-4"
                x="406"
                y="256.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="middle"
                  x="406"
                >
                  2020-11-04
                </tspan>
              </text>
            </g>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #d2d2d2; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="70"
              x2="70"
              y1="10"
              y2="230"
            />
            <text
              direction="inherit"
              dx="0"
              id="chart-axis-1-axisLabel-0"
              transform="rotate(-90,30,120)"
              x="30"
              y="117.97"
            >
              <tspan
                dx="0"
                dy="0"
                style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 40px; stroke: transparent; text-anchor: middle; fill: #252525;"
                text-anchor="middle"
                x="30"
              >
                Jobs across all clusters
              </tspan>
            </text>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="70"
                x2="65"
                y1="172.85714285714286"
                y2="172.85714285714286"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-0"
                x="55"
                y="177.82714285714286"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="end"
                  x="55"
                >
                  20
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="70"
                x2="65"
                y1="115.71428571428572"
                y2="115.71428571428572"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-1"
                x="55"
                y="120.68428571428572"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="end"
                  x="55"
                >
                  40
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="70"
                x2="65"
                y1="58.57142857142856"
                y2="58.57142857142856"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-2"
                x="55"
                y="63.54142857142856"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="end"
                  x="55"
                >
                  60
                </tspan>
              </text>
            </g>
          </g>
          <g
            clip-path="url(#victory-clip-2)"
            style="height: 100%; width: 100%; user-select: none;"
          >
            <defs>
              <clippath
                id="victory-clip-2"
              >
                <rect
                  height="220"
                  vector-effect="non-scaling-stroke"
                  width="420"
                  x="70"
                  y="10"
                />
              </clippath>
            </defs>
            <path
              d="M70,175.71428571428572L154,155.71428571428572L238,101.42857142857144L322,175.71428571428572L406,10L490,230"
              role="presentation"
              shape-rendering="auto"
              style="fill: transparent; stroke: #06c; opacity: 1; stroke-width: 2;"
            />
          </g>
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="300"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 500 300"
            width="500"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
